### PR TITLE
feat: pause ensure work object

### PR DESCRIPTION
**Pull request by CharlesQQ**
Signed-off-by: charlesQQ <charles_ali@qq.com>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes https://github.com/karmada-io/karmada/issues/1567

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```
NONE

```

### CODE REVIEW

File: pkg/apis/config/v1alpha1/resourceinterpreterwebhook_types.go [hunk 1]
```diff
@@ -116,6 +116,9 @@ const (
 	// Only necessary for those resource types that have dependencies resources and expect the dependencies be propagated
 	// together, like Deployment depends on ConfigMap/Secret.
 	InterpreterOperationInterpretDependency InterpreterOperation = "InterpretDependency"
+
+	// InterpreterOperationPauseEnsureWork indicates that karmada want to judge whether pause ensure work object or not.
+	InterpreterOperationPauseEnsureWork InterpreterOperation = "PauseEnsureWork"
```
Comment (chaunceyjiang) [inline pkg/apis/config/v1alpha1/resourceinterpreterwebhook_types.go:121]: https://github.com/karmada-io/karmada/blob/master/docs/userguide/customizing-resource-interpreter.md#built-in-interpreter

Do you need to update the document?

File: pkg/resourceinterpreter/defaultinterpreter/default.go [hunk 2]
```diff
@@ -123,3 +132,14 @@ func (e *DefaultInterpreter) ReflectStatus(object *unstructured.Unstructured) (s
...
+
+// GetPauseEnsureWorkStatus returns the status of pause ensure work object.
+func (e *DefaultInterpreter) GetPauseEnsureWorkStatus(object *unstructured.Unstructured) (pauseStatus bool) {
+	labels := object.GetLabels()
+	if labels != nil {
+		if labels[PauseEnsureWorkLabels] == True {
+			return true
+		}
+	}
+	return false
```
Comment (chaunceyjiang) [inline pkg/resourceinterpreter/defaultinterpreter/default.go:144]:
```suggestion
	return labels != nil && labels[PauseEnsureWorkLabels] == True
```

**Comment by XiShanYongYe-Chang** [comment]
Hi @CharlesQQ in favor of #5217, can we close this pr?

**Pull request closed without merging**